highlighters/main/main-highlighter.zsh

```diff
@@ -596,9 +597,18 @@ _zsh_highlight_main_highlighter_highlight_list()
     last_arg=$arg
     arg=$args[1]
     shift args
-    if (( in_alias )); then
-      (( in_alias-- ))
-      if (( in_alias == 0 )); then
+    if (( $#in_alias )); then
+      (( in_alias[-1]-- ))
```

Comment:
`arrayvar[-1]` has to walk the array (twice, IIRC), so I wonder whether inserting to the array at the start (rather than the end) would perform better.

Reply:
It also fixes the issue with 4.3.11.